nav2_smoother/include/nav2_smoother/nav2_smoother.hpp

```diff
@@ -0,0 +1,165 @@
+  /**
+   * @brief Deactivates member variables
+   *
+   * Deactivates smooth path action server, smoother, costmap and velocity
+   * publisher. Before calling deactivate state, velocity is being set to zero.
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
+  nav2_util::CallbackReturn on_deactivate(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Calls clean up states and resets member variables.
+   *
+   * Smoother and costmap clean up state is called, and resets rest of the
+   * variables
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
+  nav2_util::CallbackReturn on_cleanup(const rclcpp_lifecycle::State & state) override;
+  /**
+   * @brief Called when in Shutdown state
+   * @param state LifeCycle Node's state
+   * @return Success or Failure
+   */
+  nav2_util::CallbackReturn on_shutdown(const rclcpp_lifecycle::State & state) override;
+
+  using Action = nav2_msgs::action::SmoothPath;
+  using ActionServer = nav2_util::SimpleActionServer<Action>;
+
+  // Our action server implements the SmoothPath action
+  std::unique_ptr<ActionServer> action_server_;
+
+  /**
+   * @brief SmoothPath action server callback. Handles action server updates and
+   * spins server until goal is reached
+   *
+   * Provides global path to smoother received from action client. Local
+   * section of the path is optimized using smoother.
+   * @throw nav2_core::PlannerException
+   */
+  void smoothPlan();
+
+  /**
+   * @brief Wait for costmap to be valid with updated sensor data or repopulate
+   * after a clearing recovery. Blocks until true without timeout.
+   */
+  void waitForCostmap();
+
+  /**
+   * @brief Find the valid smoother ID name for the given request
+   *
+   * @param c_name The requested smoother name
+   * @param name Reference to the name to use for control if any valid available
+   * @return bool Whether it found a valid smoother to use
+   */
+  bool findSmootherId(const std::string & c_name, std::string & name);
+
+  // Transforms
+  std::shared_ptr<tf2_ros::Buffer> tf_;
+  std::shared_ptr<tf2_ros::TransformListener> transform_listener_;
+
+  // Publishers and subscribers
+  rclcpp_lifecycle::LifecyclePublisher<nav_msgs::msg::Path>::SharedPtr plan_publisher_;
+
+  // Smoother Plugins
+  pluginlib::ClassLoader<nav2_core::Smoother> lp_loader_;
+  SmootherMap smoothers_;
+  std::vector<std::string> default_ids_;
+  std::vector<std::string> default_types_;
+  std::vector<std::string> smoother_ids_;
+  std::vector<std::string> smoother_types_;
+  std::string smoother_ids_concat_, current_smoother_;
+
+  // Utilities
+  std::shared_ptr<nav2_costmap_2d::CostmapSubscriber> costmap_sub_;
+  std::shared_ptr<nav2_costmap_2d::FootprintSubscriber> footprint_sub_;
+  std::shared_ptr<nav2_costmap_2d::CostmapTopicCollisionChecker> collision_checker_;
+
+  rclcpp::Clock steady_clock_{RCL_STEADY_TIME};
```

Comment:
This was never gotten from the node, this need to be received from the node so the clock type can be properly set in the configure function. E.g. `node->get_clock()` & remove the default constructor you have setting it to a specific time source